### ISSUE TITLE
Require semicolon after embedded statements

### ISSUE THREAD

**Issue by amcasey** [dotnet/roslyn#6676]
e.g. this is allowed but should not be

``` C#
foreach (var proc in Process.GetProcesses())
    Console.WriteLine(proc.ProcessName) 
```

**Comment by amcasey** [comment]
FYI @tmat

**Comment by KevinH-MS** [comment]
@tmat, thoughts?

**Comment by tmat** [comment]
Yes, semicolon should be required, since foreach is not an expression.

**Comment by ivanbasov** [comment]
Fixed with #17668
